doc/source/rllib/rllib-checkpoints-and-exports.rst

```diff
@@ -0,0 +1,346 @@
+info file (``rllib_checkpoint.json``), which is always identical to the enclosing
+algorithm checkpoint version.
+
+
+How do I restore an Algorithm from a checkpoint?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Given our checkpoint path (returned by ``Algorithm.save()``), we can now
+create a completely new Algorithm instance and make it the exact same as the one we
+had stopped (and could thus no longer use) in the example above:
+
+.. literalinclude:: ../../../rllib/examples/documentation/checkpoints_and_exports.py
+    :language: python
+    :start-after: __restore-from-algo-checkpoint-begin__
+    :end-before: __restore-from-algo-checkpoint-end__
+
+
+Alternatively, you could also first create a new Algorithm instance using the
+same config that you used for the original algo, and only then call the new
+Algorithm's ``restore()`` method, passing it the checkpoint directory:
+
+.. literalinclude:: ../../../rllib/examples/documentation/checkpoints_and_exports.py
+    :language: python
+    :start-after: __restore-from-algo-checkpoint-2-begin__
+    :end-before: __restore-from-algo-checkpoint-2-end__
+
+The above procedure used to be the only way of restoring an algo, however, it is more tedious
+than using the ``from_checkpoint()`` utility as it requires an extra step and you will
+have to keep your original config stored somewhere.
+
+
+Which Algorithm checkpoint versions can I use?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+RLlib uses simple checkpoint versions (for example v0.1 or v1.0) to figure
+out how to restore an Algorithm (or a :py:class:`~ray.rllib.policy.policy.Policy`;
+see below) from a given
+checkpoint directory.
+
+From Ray 2.1 on, you can find the checkpoint version written in the
+``rllib_checkpoint.json`` file at the top-level of your checkpoint directory.
+RLlib does not use this file or information therein, it solely exists for the
+user's convenience.
+
+From Ray RLlib 2.0 and up, all checkpoint versions will be
+backward compatible, meaning some RLlib version 2.x will be able to
+handle any checkpoints created by RLlib 2.0 or any version up to 2.x.
+
+
+Multi-agent Algorithm checkpoints
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In case you are working with a multi-agent setup and have more than one
+:py:class:`~ray.rllib.policy.policy.Policy` to train inside your
+:py:class:`~ray.rllib.algorithms.algorithm.Algorithm`, you
+can create an :py:class:`~ray.rllib.algorithms.algorithm.Algorithm` checkpoint in the
+exact same way as described above and will find your individual
+:py:class:`~ray.rllib.policy.policy.Policy` checkpoints
+inside the sub-directory ``policies/``.
+
+For example:
+
+.. literalinclude:: ../../../rllib/examples/documentation/checkpoints_and_exports.py
+    :language: python
+    :start-after: __multi-agent-checkpoints-begin__
+    :end-before: __multi-agent-checkpoints-end__
+
+
+Assuming
+TODO: Restoring a multi-agent :py:class:`~ray.rllib.algorithms.algorithm.Algorithm`,
```

Comment:
just making sure to flag this TODO, although you're likely aware of it

Reply:
oh jeez, sorry.

Reply:
fixed.